tuf/ngclient/updater.py

```diff
@@ -129,6 +131,17 @@ def refresh(self) -> None:
             DownloadError: Download of a metadata file failed in some way
         """
 
+        if self.config.offline:
```

Comment:
Would that be necessary? That is just set as self.config,offline on line 106-107?

Reply:
`self.config.offline` can be modified by the user after the Updater has been initialized but before the other methods are called. This does not make a lot of sense, but technically it is possible. In other words, if we check check self.config.offline here it may well be False even though self._trusted_set has been initialized as offline.

The current test case seems to be a good example of this.